IshiHisashi/GlucoFit
# feat: JDEV-66 - Create Home screen

## Description of this change
- add basic contents and layout on Home screen. 
- add chart library (Victory Native)
- integrate APIs to show contents on Home screen (the line chart and the logs table. the bar chart on the bottom is still hard coded. )
- make sure Home screen is updated after creating a log 
- there's timestamp inconsistency issue involving frontend and backend. This will be solved in another PR. 
- styling is NOT fully prioritized. 

## Steps to test
1. `cd frontend`, `npm install`, and start the app on an emulator
2. you will see home screen contents. The line chart section and the logs table are coming from DB data. The weekly bar chart on the bottom is hardcoded. 
3. you can create a new glucose log. If the date and time of the log are within "today", you'll see the newly-created log data on Home screen after creating it. 
### before creating a new glucose log

<img src="https://github.com/user-attachments/assets/c81521e1-40dc-41c0-b2f9-e25f4db35f4b" width="150px" />
<img src="https://github.com/user-attachments/assets/263833ee-9d1a-4096-a729-5aca5188bffb" width="150px" />

### after creating a new glucose log

<img src="https://github.com/user-attachments/assets/0447c420-be5f-4e1d-a06a-298861277507" width="150px" />
<img src="https://github.com/user-attachments/assets/3e0c0081-0b43-4a11-b751-39b72049c519" width="150px" />

## Ticket link
https://team-jupiter-capstone.atlassian.net/browse/JDEV-66
